src/modal/Modal.tsx

```diff
@@ -160,15 +166,24 @@ const Modal: React.FC = () => {
                 placeholder="Derivation Path"
                 value={ledgerDerivationPath}
                 onChange={handleDerivationPathChange}
+                readOnly={isLoading}
               />
               {ledgerError && <p className="error">{ledgerError}</p>}
             </div>
             <div className="derivation-paths--actions">
-              <button className="left-button" onClick={handleDismissClick}>
+              <button
+                className="left-button"
+                onClick={handleDismissClick}
+                disabled={isLoading}
+              >
                 Dismiss
               </button>
-              <button className="right-button" onClick={handleConnectClick}>
-                Connect
+              <button
+                className="right-button"
+                onClick={handleConnectClick}
+                disabled={isLoading}
+              >
+                {isLoading ? "Loading..." : "Connect"}
```

Comment:
I know I previously suggested "Loading..." but maybe "Connecting..." would be better here until we have a spinner. Not sure if it causes any reflow problems with it being a longer string?